appinventor/blocklyeditor/src/component_database.js

```diff
@@ -504,6 +514,45 @@ Blockly.ComponentDatabase.prototype.processOption = function(option) {
   };
 }
 
+/**
+ * Processes data defining an asset filter (from simple_components.json) and
+ * returns a HelperKey pointing to the filter.
+ * @param {!Object} data The data defining the filter.
+ * @return {!HelperKey} The key associated with the filter.
+ */
+Blockly.ComponentDatabase.prototype.processAssetHelper = function(data) {
+  var filter = data.filter;
+  if (!filter || filter.length == 0) {
+    return {
+      type: "ASSET",
+      key: null
+    }
+  }
+  // We have to do this because js is very restrictive with array equality.
+  var findIndex = function(acc, cur, idx) {
```

Comment:
I think the recommended best practice is to just use `function findIndex(..)` instead of assigning the anonymous function to a variable.